### ISSUE TITLE
node: implement claim support

### ISSUE THREAD

**Issue by jcorbin** [uber/tchannel#126]
- [ ] support operation claim
- [ ] claim frame parser
- [ ] internal handler support
- [ ] API through TChannelConnection?
- [ ] API through TChannel?

**Comment by kriskowal** [comment]
We will need a data structure for collecting tombstones for claimed requests, keyed by tracing information. We should also consider adding a ttl to cancel and claim frames so that we know when we can collect these tombstones.

**Comment by ShanniLi** [comment]
@kriskowal agreed. The claim requests can overflow. We may add a ttl to the frame or we may define a local policy to retire claim requests. E.g., a combination of FIFO and ttl for the queue that keeps the claims.
